tabular/src/autogluon/tabular/models/abstract/abstract_model.py

```diff
@@ -118,11 +118,22 @@ def can_infer(self) -> bool:
     def is_fit(self) -> bool:
         return self.model is not None
 
+    # TODO: v0.1 update to be aligned with _set_default_auxiliary_params(), add _get_default_params()
     def _set_default_params(self):
         pass
 
     def _set_default_auxiliary_params(self):
+        """
+        Sets the default aux parameters of the model.
+        This method should not be extended by inheriting models, instead extend _get_default_auxiliary_params.
+        """
         # TODO: Consider adding to get_info() output
+        default_auxiliary_params = self._get_default_auxiliary_params()
+        for key, value in default_auxiliary_params.items():
+            self._set_default_param_value(key, value, params=self.params_aux)
+
+    def _get_default_auxiliary_params(self) -> dict:
+        """Dictionary of auxiliary parameters that dictate various model-agnostic logic."""
```

Comment:
I'd provide some examples in docstring.

"logic, such as: which dtypes are columns that are ignored by the model, or how much memory the model is allowed to use."

Reply:
1) if this method is expected to be extended in child classes, should it be marked as internal?
2) since this is effectively is implemented in each model, it's probably a good idea to add example in docstring.

Reply:
1. _ just means that it shouldn't be called outside of the object itself. I don't think it has any relation to user extensions. For example, we encourage users to extend _fit, _predict_proba, and _preprocess

2. Perhaps example would live best in the example custom model code? Not sure it belongs in the docstring.

Reply:
Added extra examples to docstring